test/util/client/azure/azure.go

```diff
@@ -0,0 +1,109 @@
+//+build e2e
+
+package azure
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	"github.com/kelseyhightower/envconfig"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/log"
+
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-05-01/resources"
+	"github.com/onsi/ginkgo/config"
+
+	"github.com/openshift/openshift-azure/pkg/util/azureclient"
+	sdk "github.com/openshift/openshift-azure/pkg/util/azureclient/osa-go-sdk/services/containerservice/mgmt/2018-09-30-preview/containerservice"
+)
+
+var (
+	fakeRe = regexp.MustCompile("Fake")
+	realRe = regexp.MustCompile("Real")
+)
+
+type Config struct {
+	SubscriptionID  string   `envconfig:"AZURE_SUBSCRIPTION_ID" required:"true"`
+	TenantID        string   `envconfig:"AZURE_TENANT_ID" required:"true"`
+	Region          string   `envconfig:"AZURE_REGION" required:"true"`
+	ClientID        string   `envconfig:"AZURE_CLIENT_ID" required:"true"`
+	ClientSecret    string   `envconfig:"AZURE_CLIENT_SECRET" required:"true"`
+	ResourceGroup   string   `envconfig:"RESOURCEGROUP" required:"true"`
+	AcceptLanguages []string `envconfig:"ACCEPT_LANGUAGES" default:"en-us"`
+}
+
+var conf Config
+
+func init() {
+	err := envconfig.Process("", &conf)
+	if err != nil {
+		panic(err)
+	}
+}
+
+type Client struct {
+	gc    resources.GroupsClient
+	rpc   sdk.OpenShiftManagedClustersClient
+	ssc   azureclient.VirtualMachineScaleSetsClient
+	ssvmc azureclient.VirtualMachineScaleSetVMsClient
+	ssec  azureclient.VirtualMachineScaleSetExtensionsClient
+	appsc azureclient.ApplicationsClient
+
+	resourceGroup string
+	location      string
+	logger        *logrus.Entry
+	ctx           context.Context
+}
+
+func NewClient() *Client {
+	authorizer, err := azureclient.NewAuthorizer(conf.ClientID, conf.ClientSecret, conf.TenantID)
+	if err != nil {
+		panic(err)
+	}
+	subID := conf.SubscriptionID
+	gc := resources.NewGroupsClient(subID)
+	gc.Authorizer = authorizer
+
+	var rpc sdk.OpenShiftManagedClustersClient
+	focus := []byte(config.GinkgoConfig.FocusString)
+	switch {
+	case fakeRe.Match(focus):
+		fmt.Println("Creating a cluster using the fake resource provider")
+		rpc = sdk.NewOpenShiftManagedClustersClientWithBaseURI("http://localhost:8080", subID)
```

Comment:
There is a const in pkg/fakerp about the location the fake RP is going to serve. Expose that if it's not exposed already and use it here.

Reply:
Disregard, this is not a thing anymore.